File: Single-Linked-List/insert_node_at_position.c
/* File Name: insert_at_position.c */

#include <stdio.h>
#include <stdlib.h>

// Definition of Node
struct Node {
    int data;
    struct Node *next;
};

// Function to insert node at a given position
struct Node* insert_pos(struct Node *head, int pos) {
    struct Node *newNode, *temp;
    int i;

    newNode = (struct Node*)malloc(sizeof(struct Node));

    if (newNode == NULL) {
        printf("Memory allocation failed\n");
        return head;
    }

    printf("Enter data for new node: ");
    scanf("%d", &newNode->data);

    // Insert at beginning if position is 1
    if (pos == 1) {
        newNode->next = head;
        head = newNode;
        return head;
    }

    temp = head;
    for (i = 1; i < pos - 1 && temp != NULL; i++) {
        temp = temp->next;
    }

    if (te

File: Single-Linked-List/insert_node_at_position.py
# File Name: insert_at_position.py

# Node class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Singly Linked List class
class SinglyLinkedList:
    def __init__(self):
        self.head = None

    # Method to insert at a given position
    def insert_pos(self, pos):
        value = int(input("Enter data for new node: "))
        new_node = Node(value)

        # Insert at beginning
        if pos == 1:
            new_node.next = self.head
            self.head = new_node
            return

        temp = self.head
        for _ in range(pos - 2):
            if temp is None:
                print("Invalid position")
                return
            temp = temp.next

        new_node.next = temp.next
        temp.next = new_node
